src/main/java/seedu/address/logic/commands/ViewCommand.java

```diff
@@ -0,0 +1,82 @@
+package seedu.address.logic.commands;
+
+import static java.util.Objects.requireNonNull;
+import static seedu.address.logic.Messages.MESSAGE_PATIENT_LISTED_SUCCESS;
+
+import seedu.address.commons.core.LogsCenter;
+import seedu.address.commons.util.ToStringBuilder;
+import seedu.address.model.Model;
+import seedu.address.model.patient.IcNumber;
+import seedu.address.model.patient.PatientWithIcNumberPredicate;
+import seedu.address.model.patient.exceptions.PatientWithFieldNotFoundException;
+
+import java.util.logging.Logger;
+
+/**
+ * Views the Patient with a certain {@Code IcNumber}
+ */
+public class ViewCommand extends Command {
+
+    public static final String COMMAND_WORD = "view";
+
+    public static final String MESSAGE_USAGE =
+        COMMAND_WORD + ": View the Patient with IcNumber\n" + "Parameters: IC_NUMBER\n" + "Example: " + COMMAND_WORD
+            + " i/t1234567j";
+
+    private final PatientWithIcNumberPredicate predicate;
+    private final IcNumber icNumberToFind;
+    private final Logger logger = LogsCenter.getLogger(getClass());
+
+    /**
+     * Creates a ViewCommand object to be executed
+     *
+     * @param predicate      The predicate used to filter Patients
+     * @param icNumberToFind The icNumber entered by user input to be find
+     */
+    public ViewCommand(PatientWithIcNumberPredicate predicate, IcNumber icNumberToFind)
+        throws PatientWithFieldNotFoundException {
+        if (icNumberToFind == null) {
+            throw new PatientWithFieldNotFoundException("Multiplicity violated. Each patient has 1 non-null IcNumber.");
+        }
+        this.predicate = predicate;
+        this.icNumberToFind = icNumberToFind;
+    }
+
+    /**
+     * Executes the {@Code executes ViewCommand} using the current model and return {@Code CommandResult}
+     *
+     * @param model {@code Model} which the command should operate on.
+     * @return CommandResult object
+     * @throws PatientWithFieldNotFoundException If user enters a field not present in any existing patients
+     */
+    @Override
+    public CommandResult execute(Model model) throws PatientWithFieldNotFoundException {
```

Comment:
Hmm just to clarify, your implementation of the view command works like find by name, right? In that it filters the list for the patient with the matching ic number? In the event that a patient has yet to be selected in the ui's displayed list (this happens when the app is first opened and no patient is selected yet, after selecting I believe one patient on the list will always be selected), is the matching patient's record still automatically displayed on the right side of the ui? Or is it still blank (no patient selected). 
If the case is that the patient's record is not automatically displayed after entering the command, it is still totally fine but the user will still have to click and select the card from the ui list, contradicting our 'no clicks/gui interactions' 'rule' (for the first time only). This is honestly quite a minor thing HAHA

Reply:
Hi yes the patient will be displayed on the right side even on startup with no patient displayed. And the user would have to click on it to view details as well. Perhaps if we want to make it more seamless, ie show straightaway, i could get the UI in charge to integrate!